<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Devon Allies">

<title>Pathogen Identification from Clinical Metagenomics Data â€“ Data Science Projects</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ed96de9b727972fe78a7b5d16c58bf87.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-5dace33b8eeff415c0721a143d25f203.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data Science Projects</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../projects.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About Me</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#aim" id="toc-aim" class="nav-link active" data-scroll-target="#aim">Aim:</a></li>
  <li><a href="#data-acquisition" id="toc-data-acquisition" class="nav-link" data-scroll-target="#data-acquisition">Data Acquisition:</a></li>
  <li><a href="#quality-control-of-samples" id="toc-quality-control-of-samples" class="nav-link" data-scroll-target="#quality-control-of-samples">Quality Control of Samples:</a>
  <ul class="collapse">
  <li><a href="#fastq-statistics" id="toc-fastq-statistics" class="nav-link" data-scroll-target="#fastq-statistics">Fastq Statistics:</a></li>
  </ul></li>
  <li><a href="#bam-flagstat-statistics" id="toc-bam-flagstat-statistics" class="nav-link" data-scroll-target="#bam-flagstat-statistics">BAM Flagstat Statistics:</a></li>
  <li><a href="#host-depletion" id="toc-host-depletion" class="nav-link" data-scroll-target="#host-depletion">Host Depletion:</a></li>
  <li><a href="#de-nova-assembly" id="toc-de-nova-assembly" class="nav-link" data-scroll-target="#de-nova-assembly">De Nova Assembly:</a></li>
  <li><a href="#basic-local-alignment-search-tool" id="toc-basic-local-alignment-search-tool" class="nav-link" data-scroll-target="#basic-local-alignment-search-tool">Basic Local Alignment Search Tool:</a></li>
  <li><a href="#sample-identification" id="toc-sample-identification" class="nav-link" data-scroll-target="#sample-identification">Sample Identification:</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion:</a></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference">Reference:</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Pathogen Identification from Clinical Metagenomics Data</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Devon Allies </p>
          </div>
  </div>
    
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    <p>This analysis focuses on identifying a potential pathogen from clinical metagenomics data. The workflow includes quality control of sequencing reads, host depletion, de novo assembly, and classification using BLAST. The results highlight the presence of clinically relevant pathogens in the sample.</p>
  </div>
</div>


</header>


<section id="aim" class="level2">
<h2 class="anchored" data-anchor-id="aim">Aim:</h2>
<p>The objective of this project was to develop and execute a bioinformatics workflow to identify potential pathogens from clinical metagenomics data. The workflow encompassed several key steps, including quality control of raw sequencing reads, host depletion to remove human DNA, de novo assembly of unmapped reads, and classification using BLAST. The ultimate goal was to determine the presence of clinically relevant pathogens in the sample. By performing host depletation and classification, I was able to demonstrate the ability to isolate the microbial signal from a Bronchoalveolar lavage (BAL) sample.</p>
</section>
<section id="data-acquisition" class="level2">
<h2 class="anchored" data-anchor-id="data-acquisition">Data Acquisition:</h2>
<p>The raw sequencing data for this project was accessed from the NCBI SRA (BioProject: PRJNA1327646), originating from a published study on COPD microbiomes <span class="citation" data-cites="Chen2025">(<a href="#ref-Chen2025" role="doc-biblioref">Chen et al., 2025</a>)</span>. The authors of the study specified that the data deposited in the SRA had already undergone in silico human host DNA removal prior to submission.</p>
<p>To ensure data integrity and quantify any residual human reads, a Quality Control (QC) step was performed. Raw reads were aligned against the human reference genome (hg38) using BWA <span class="citation" data-cites="bwa">(<a href="#ref-bwa" role="doc-biblioref">Li and Durbin, 2009</a>)</span>.</p>
</section>
<section id="quality-control-of-samples" class="level2">
<h2 class="anchored" data-anchor-id="quality-control-of-samples">Quality Control of Samples:</h2>
<section id="fastq-statistics" class="level3">
<h3 class="anchored" data-anchor-id="fastq-statistics">Fastq Statistics:</h3>
<p>The <code>seqkit stats</code> <span class="citation" data-cites="seqkit">(<a href="#ref-seqkit" role="doc-biblioref">Shen, Sipos and Zhao, 2024</a>)</span> output describes the raw sequencing data before alignment. The command is used to generate a tabular statistical summary of <code>FASTA/Q</code> files. By default, it automatically detects the sequence type of the files.</p>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/seqkit</span> stats <span class="va">${dataDir}</span>/reads/SRR35359600_<span class="pp">*</span>.fastq</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-fq-stats" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-fq-stats-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Statistics of fastq files
</figcaption>
<div aria-describedby="tbl-fq-stats-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>processed files:  0 / 2 [------] ETA: 0s
[1A[Jprocessed files:  0 / 2 [------] ETA: 0s
[1A[Jprocessed files:  2 / 2 [======] ETA: 0s
[1A[Jprocessed files:  2 / 2 [] ETA: 0s. done
file                                        format  type   num_seqs      sum_len  min_len  avg_len  max_len
../../../../data/reads/SRR35359600_1.fastq  FASTQ   DNA   1,246,647  182,039,361       15      146      150
../../../../data/reads/SRR35359600_2.fastq  FASTQ   DNA   1,246,647  181,919,031       15    145.9      150</code></pre>
</div>
</div>
</figure>
</div>
</section>
</section>
<section id="bam-flagstat-statistics" class="level2">
<h2 class="anchored" data-anchor-id="bam-flagstat-statistics">BAM Flagstat Statistics:</h2>
<p>The <code>BAM flagstat</code> statistic summarizes the results after the raw reads have been aligned / mapped to a reference genome.</p>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/sort/SRR35359600.sorted.bam"</span> .sorted.bam<span class="va">)</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing sample: </span><span class="va">$sample</span><span class="st">"</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/samtools</span> flagstat <span class="va">${dataDir}</span>/sort/SRR35359600.sorted.bam</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-bam-flagstat" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-bam-flagstat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Statistics of BAM files
</figcaption>
<div aria-describedby="tbl-bam-flagstat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>------------------------------
Processing sample: SRR35359600
-------------------------------
2627660 + 0 in total (QC-passed reads + QC-failed reads)
2493294 + 0 primary
0 + 0 secondary
134366 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
1637803 + 0 mapped (62.33% : N/A)
1503437 + 0 primary mapped (60.30% : N/A)
2493294 + 0 paired in sequencing
1246647 + 0 read1
1246647 + 0 read2
1169744 + 0 properly paired (46.92% : N/A)
1496222 + 0 with itself and mate mapped
7215 + 0 singletons (0.29% : N/A)
241104 + 0 with mate mapped to a different chr
113350 + 0 with mate mapped to a different chr (mapQ&gt;=5)</code></pre>
</div>
</div>
</figure>
</div>
<ul>
<li><p><code>total</code>: The total number of reads processed, separated into those passing Quality Control (QC) and those failing.</p></li>
<li><p><code>mapped</code>: The percentage of reads that successfully aligned to the reference genome. This is critical for assessing the efficiency of the alignment process and the relevance of the reference genome.</p></li>
<li><p><code>paired in sequencing</code>: Confirms all reads were generated from paired-end sequencing, which is important for downstream analyses that rely on paired-end data.</p></li>
<li><p><code>properly paired</code>: Reads that aligned to the reference genome in the expected orientation and distance. A high percentage indicates good library preparation and alignment quality.</p></li>
<li><p><code>singletons</code>: Reads that did not have a properly paired mate.</p></li>
</ul>
</section>
<section id="host-depletion" class="level2">
<h2 class="anchored" data-anchor-id="host-depletion">Host Depletion:</h2>
<p>Host depletion is a critical step in viral metagenomics to enrich for viral sequences by removing host-derived reads. This process can be achieved through various methods, including computational approaches that filter out host sequences based on alignment to a reference genome or using specialized tools designed for host depletion.</p>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the paths for files and environment</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/sort/SRR35359600.sorted.bam"</span> .sorted.bam<span class="va">)</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing sample: </span><span class="va">$sample</span><span class="st">"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/samtools</span> collate <span class="at">-u</span> <span class="at">-O</span> <span class="va">${dataDir}</span>/sort/SRR35359600.sorted.bam <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/samtools</span> fastq <span class="at">-f4</span> <span class="at">-1</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/</span><span class="va">${sample}</span><span class="st">_R1.fq"</span> <span class="at">-2</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/</span><span class="va">${sample}</span><span class="st">_R2.fq"</span> <span class="at">-s</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/</span><span class="va">${sample}</span><span class="st">_singleton.fq"</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-host-depletion" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-host-depletion-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Removing unmapped reads
</figcaption>
<div aria-describedby="tbl-host-depletion-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>------------------------------
Processing sample: SRR35359600
-------------------------------
[M::bam2fq_mainloop] discarded 7215 singletons
[M::bam2fq_mainloop] processed 989857 reads</code></pre>
</div>
</div>
</figure>
</div>
<p>Host depletion is a critical step in clinical metagenomics because clinical samples (like blood, tissue, or respirator fluid) are typically dominated by human DNA, which can constitute 90% - 99% of the sequencing data. By removing the host background, sensitivity is increased, it enables accurate De Novo Assembly, and improves downstream analysis. The singletons are placed in a <code>singleton.fq</code> file.</p>
<p>While a notable percentage indicated in <a href="#tbl-bam-flagstat" class="quarto-xref">Table&nbsp;2</a>, 62.33%, of the reads still aligned to the human genome, the remaining non-host reads are sufficient for downstream classification.</p>
</section>
<section id="de-nova-assembly" class="level2">
<h2 class="anchored" data-anchor-id="de-nova-assembly">De Nova Assembly:</h2>
<p>Megahit <span class="citation" data-cites="megahit">(<a href="#ref-megahit" role="doc-biblioref">Li et al., 2016</a>)</span> is a fast and memory-efficient de novo assembler for large and complex metagenomics data sets. It uses succinct de Bruijn graphs to represent the input data, which significantly reduces memory usage compared to traditional de Bruijn graph representations. Megahit is designed to handle large-scale metagenomic datasets, making it suitable for assembling genomes from environmental samples with high diversity and complexity.</p>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the paths for files and environment</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/SRR35359600"</span> _R1.fq<span class="va">)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="va">r1</span><span class="op">=</span><span class="va">${dataDir}</span>/unmapped/<span class="va">${sample}</span>_R1.fq</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="va">r2</span><span class="op">=</span><span class="va">${dataDir}</span>/unmapped/<span class="va">${sample}</span>_R2.fq</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="va">singleton</span><span class="op">=</span><span class="va">${dataDir}</span>/unmapped/<span class="va">${sample}</span>_singleton.fq</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing sample: </span><span class="va">$sample</span><span class="st">"</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/megahit</span> <span class="at">-1</span> <span class="va">$r1</span> <span class="at">-2</span> <span class="va">$r2</span> <span class="at">-r</span> <span class="va">$singleton</span> <span class="at">-o</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/megahit/</span><span class="va">${sample}</span><span class="st">"</span> <span class="at">--out-prefix</span> <span class="va">$sample</span> <span class="at">-f</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-denova" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-denova-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: De Nova Assembly uisng Megahits
</figcaption>
<div aria-describedby="tbl-denova-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>------------------------------
Processing sample: SRR35359600
-------------------------------
2026-02-05 14:24:59 - MEGAHIT v1.2.9
2026-02-05 14:24:59 - Using megahit_core with POPCNT and BMI2 support
2026-02-05 14:24:59 - Convert reads to binary library
2026-02-05 14:25:00 - b'INFO  sequence/io/sequence_lib.cpp  :   75 - Lib 0 (/mnt/069ABB979ABB822B/learning/data/unmapped/SRR35359600_R1.fq,/mnt/069ABB979ABB822B/learning/data/unmapped/SRR35359600_R2.fq): pe, 982642 reads, 150 max length'
2026-02-05 14:25:00 - b'INFO  sequence/io/sequence_lib.cpp  :   75 - Lib 1 (/mnt/069ABB979ABB822B/learning/data/unmapped/SRR35359600_singleton.fq): se, 7215 reads, 150 max length'
2026-02-05 14:25:00 - b'INFO  utils/utils.h                 :  152 - Real: 1.0382\tuser: 0.6381\tsys: 0.1387\tmaxrss: 85204'
2026-02-05 14:25:00 - k-max reset to: 141 
2026-02-05 14:25:00 - Start assembly. Number of CPU threads 8 
2026-02-05 14:25:00 - k list: 21,29,39,59,79,99,119,141 
2026-02-05 14:25:00 - Memory used: 45253907251
2026-02-05 14:25:00 - Extract solid (k+1)-mers for k = 21 
2026-02-05 14:25:09 - Build graph for k = 21 
2026-02-05 14:25:13 - Assemble contigs from SdBG for k = 21
2026-02-05 14:25:22 - Local assembly for k = 21
2026-02-05 14:25:28 - Extract iterative edges from k = 21 to 29 
2026-02-05 14:25:29 - Build graph for k = 29 
2026-02-05 14:25:31 - Assemble contigs from SdBG for k = 29
2026-02-05 14:25:40 - Local assembly for k = 29
2026-02-05 14:25:46 - Extract iterative edges from k = 29 to 39 
2026-02-05 14:25:47 - Build graph for k = 39 
2026-02-05 14:25:49 - Assemble contigs from SdBG for k = 39
2026-02-05 14:25:59 - Local assembly for k = 39
2026-02-05 14:26:08 - Extract iterative edges from k = 39 to 59 
2026-02-05 14:26:08 - Build graph for k = 59 
2026-02-05 14:26:10 - Assemble contigs from SdBG for k = 59
2026-02-05 14:26:17 - Local assembly for k = 59
2026-02-05 14:26:24 - Extract iterative edges from k = 59 to 79 
2026-02-05 14:26:25 - Build graph for k = 79 
2026-02-05 14:26:26 - Assemble contigs from SdBG for k = 79
2026-02-05 14:26:30 - Local assembly for k = 79
2026-02-05 14:26:38 - Extract iterative edges from k = 79 to 99 
2026-02-05 14:26:39 - Build graph for k = 99 
2026-02-05 14:26:40 - Assemble contigs from SdBG for k = 99
2026-02-05 14:26:44 - Local assembly for k = 99
2026-02-05 14:26:51 - Extract iterative edges from k = 99 to 119 
2026-02-05 14:26:52 - Build graph for k = 119 
2026-02-05 14:26:53 - Assemble contigs from SdBG for k = 119
2026-02-05 14:26:56 - Local assembly for k = 119
2026-02-05 14:27:04 - Extract iterative edges from k = 119 to 141 
2026-02-05 14:27:05 - Build graph for k = 141 
2026-02-05 14:27:05 - Assemble contigs from SdBG for k = 141
2026-02-05 14:27:08 - Merging to output final contigs 
2026-02-05 14:27:09 - 9116 contigs, total 7205375 bp, min 210 bp, max 16159 bp, avg 790 bp, N50 908 bp
2026-02-05 14:27:09 - ALL DONE. Time elapsed: 129.526783 seconds </code></pre>
</div>
</div>
</figure>
</div>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the paths for files and environment</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/SRR35359600"</span> _R1.fq<span class="va">)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/seqkit</span> stats <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span>.contigs.fa</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-megahit-stats" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-megahit-stats-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5: Statistics of Megahit Assembly
</figcaption>
<div aria-describedby="tbl-megahit-stats-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>file                                                         format  type  num_seqs    sum_len  min_len  avg_len  max_len
../../../../data/megahit/SRR35359600/SRR35359600.contigs.fa  FASTA   DNA      9,116  7,205,375      210    790.4   16,159</code></pre>
</div>
</div>
</figure>
</div>
</section>
<section id="basic-local-alignment-search-tool" class="level2">
<h2 class="anchored" data-anchor-id="basic-local-alignment-search-tool">Basic Local Alignment Search Tool:</h2>
<p>Blast is the gold standard for comparing a query sequence (DNA, RNA, or Protein) against a massive database of known sequences to find regions of similarity</p>
<div class="cell">
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|label: tbl-blast</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#|tbl-cap: "Output of blast result"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/SRR35359600"</span> _R1.fq<span class="va">)</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/makeblastdb</span> <span class="at">-out</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span> <span class="at">-in</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span>.contigs.fa <span class="at">-dbtype</span> nucl <span class="at">-parse_seqids</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastdbcmd</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span> <span class="at">-entry</span> all <span class="at">-outfmt</span> <span class="st">"%l %a"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/megahit/</span><span class="va">${sample}</span><span class="st">/</span><span class="va">${sample}</span><span class="st">_blast.txt"</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span>_blast.txt <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-rn</span> <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>

Building a new DB, current time: 02/06/2026 09:52:51
New DB name:   /mnt/069ABB979ABB822B/learning/data/megahit/SRR35359600/SRR35359600
New DB title:  ../../../../data/megahit/SRR35359600/SRR35359600.contigs.fa
Sequence type: Nucleotide
Deleted existing Nucleotide BLAST database named /mnt/069ABB979ABB822B/learning/data/megahit/SRR35359600/SRR35359600
Keep MBits: T
Maximum file size: 3000000000B
Adding sequences from FASTA; added 9116 sequences in 0.204801 seconds.


16159 k141_9107
15399 k141_6789
15193 k141_7744
9747 k141_191
8971 k141_6695
8401 k141_4630
8326 k141_89
8324 k141_2178
7369 k141_8263
6914 k141_3816</code></pre>
</div>
</div>
</section>
<section id="sample-identification" class="level2">
<h2 class="anchored" data-anchor-id="sample-identification">Sample Identification:</h2>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/SRR35359600"</span> _R1.fq<span class="va">)</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastdbcmd</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span> <span class="at">-entry</span> k141_9107 <span class="op">&gt;</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/megahit/</span><span class="va">${sample}</span><span class="st">/k141_9107.fa"</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastdbcmd</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span> <span class="at">-entry</span> k141_6789 <span class="op">&gt;</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/megahit/</span><span class="va">${sample}</span><span class="st">/k141_6789.fa"</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastdbcmd</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span> <span class="at">-entry</span> k141_7744 <span class="op">&gt;</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/megahit/</span><span class="va">${sample}</span><span class="st">/k141_7744.fa"</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Blast k141_9107</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing contig: k141_9107"</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastn</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/databases/ref_prok_rep_genomes <span class="at">-query</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/k141_9107.fa <span class="at">-outfmt</span> <span class="st">"6 qseqid pident evalue length stitle"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$2 &gt;= 99.0 &amp;&amp; $4 &gt;= 1000'</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Blast k141_6789</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing contig: k141_6789"</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastn</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/databases/ref_prok_rep_genomes <span class="at">-query</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/k141_6789.fa <span class="at">-outfmt</span> <span class="st">"6 qseqid pident evalue length stitle"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$2 &gt;= 99.0 &amp;&amp; $4 &gt;= 1000'</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Blast k141_7744</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"------------------------------"</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"Processing contig: k141_7744"</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"-------------------------------"</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/blastn</span> <span class="at">-db</span> <span class="va">${dataDir}</span>/databases/ref_prok_rep_genomes <span class="at">-query</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/k141_7744.fa <span class="at">-outfmt</span> <span class="st">"6 qseqid pident evalue length stitle"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'$2 &gt;= 99.0 &amp;&amp; $4 &gt;= 1000'</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-idsample" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-idsample-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;6: Top BLAST Hits for Sample
</figcaption>
<div aria-describedby="tbl-idsample-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>------------------------------
Processing contig: k141_9107
-------------------------------
k141_9107   99.409  0.0 5757    Acinetobacter parvus strain CCM 7030 tig00001935, whole genome shotgun sequence
k141_9107   99.274  0.0 1102    Acinetobacter ursingii NIPH 706 acLZz-supercont1.16, whole genome shotgun sequence
k141_9107   99.186  0.0 1106    Acinetobacter thermotolerans strain ANC 7924 chromosome, complete genome
k141_9107   99.008  0.0 1109    Acinetobacter baumannii strain ATCC 19606 chromosome, complete genome
k141_9107   99.005  0.0 1106    Acinetobacter baumannii strain ATCC 19606 chromosome, complete genome
k141_9107   99.092  0.0 1101    Acinetobacter baumannii strain ATCC 19606 chromosome, complete genome
------------------------------
Processing contig: k141_6789
-------------------------------
k141_6789   99.864  0.0 5156    Faucicola atlantae strain NCTC11091, whole genome shotgun sequence
------------------------------
Processing contig: k141_7744
-------------------------------</code></pre>
</div>
</div>
</figure>
</div>
<p>The <code>BLAST</code> results indicate a statistically definitive match between the query sequence and the subject database. The alignment parameters - specifically the 99.409% percent identity, the 5757 basepair alignmnent length, and the E-value of 0.0 - provides evidence of high-confidence sequence homology.</p>
<ul>
<li><p><code>Percent Identity</code>: The high degree of nucleotide conservation suggests that the query and subject sequences are nearly identical. This level of similarity is characteristic of sequences derived from the same species or highly conserved orthologous rehions within closely related strains.</p></li>
<li><p><code>Alignment Length</code>: The substantial length of the aligned region bolsters the validity of the match.</p></li>
<li><p><code>Expect Value</code>: An E-value of 0.0 represents a probability that is indistinguishable from zero. It signifies that the likelyhood of observing such a high-scoring alignment by random chance within the search space is non-existent.</p></li>
</ul>
<p>The verified non-host reads were assembled into contigs using <code>megahit</code> <span class="citation" data-cites="megahit">(<a href="#ref-megahit" role="doc-biblioref">Li et al., 2016</a>)</span> and subsequently classified via <code>BLAST</code> <span class="citation" data-cites="blast">(<a href="#ref-blast" role="doc-biblioref">Camacho et al., 2009</a>)</span>. The analysis yielded a high-confidence identification based on the alignment of contig <code>k141_9107</code></p>
<details class="code-fold">
<summary>Show code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="va">envDir</span><span class="op">=</span><span class="st">"/home/devonallies/micromamba/envs/bioinfo/bin"</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="va">dataDir</span><span class="op">=</span><span class="st">"../../../../data"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${dataDir}</span><span class="st">/unmapped/SRR35359600"</span> _R1.fq<span class="va">)</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="va">${envDir}</span><span class="ex">/amrfinder</span> <span class="at">-n</span> <span class="va">${dataDir}</span>/megahit/<span class="va">${sample}</span>/<span class="va">${sample}</span>.contigs.fa <span class="at">--database</span> <span class="va">${dataDir}</span>/databases/armfinder/amr_db/2026-01-21.1/ <span class="at">--plus</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div id="tbl-amrfinder" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-amrfinder-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;7: AMRFinder Results for Sample
</figcaption>
<div aria-describedby="tbl-amrfinder-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output cell-output-stdout">
<pre><code>Running: /home/devonallies/micromamba/envs/bioinfo/bin/amrfinder -n ../../../../data/megahit/SRR35359600/SRR35359600.contigs.fa --database ../../../../data/databases/armfinder/amr_db/2026-01-21.1/ --plus
Software directory: /home/devonallies/micromamba/envs/bioinfo/bin/
Software version: 4.2.5
Database directory: /mnt/069ABB979ABB822B/learning/data/databases/armfinder/amr_db/2026-01-21.1
Database version: 2026-01-21.1
AMRFinder translated nucleotide search
  - include -O ORGANISM, --organism ORGANISM option to add mutation searches and suppress common proteins
Running blastx
Making report
Protein id  Contig id   Start   Stop    Strand  Element symbol  Element name    Scope   Type    Subtype Class   Subclass    Method  Target length   Reference sequence length   % Coverage of reference % Identity to reference Alignment length    Closest reference accession Closest reference name  HMM accession   HMM description
NA  k141_2664   760 1617    +   blaTEM-116  broad-spectrum class A beta-lactamase TEM-116   core    AMR AMR BETA-LACTAM BETA-LACTAM ALLELEX 286 286 100.00  100.00  286 WP_000027050.1  broad-spectrum class A beta-lactamase TEM-116   NA  NA
amrfinder took 47 seconds to complete</code></pre>
</div>
</div>
</figure>
</div>
<p><code>AMRFINDER</code> identified the gene <code>blaTEM-116</code>, which is a broad-spectrum class A beta-lactamase. This enzyme breaks down beta-lactam antibiotics.</p>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion:</h2>
<p>The identification of <em>Acinetobacter parvus</em>, is a gram-negative aerobic bacterium notable for forming small colonies in cultures, with the metrics displayed in <a href="#tbl-idsample" class="quarto-xref">Table&nbsp;6</a>, indicates a strong result.</p>
</section>
<section id="reference" class="level2">




</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">Reference:</h2><div id="refs" class="references csl-bib-body" role="list">
<div id="ref-blast" class="csl-entry" role="listitem">
Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J., Bealer, K. and Madden, T.L., 2009. BLAST+: Architecture and applications. <em>BMC Bioinformatics</em>, 10(1), p.421. https://doi.org/<a href="https://doi.org/10.1186/1471-2105-10-421">10.1186/1471-2105-10-421</a>.
</div>
<div id="ref-Chen2025" class="csl-entry" role="listitem">
Chen, G., Wiegand, C., Willett, A., Herr, C., MÃ¼ller, R., Bals, R. and Kalinina, O.V., 2025. Comparative metagenomic analysis on <span>COPD</span> and health control samples reveals taxonomic and functional motifs. <em>Frontiers in Microbiology</em>, 16, p.1636322. https://doi.org/<a href="https://doi.org/10.3389/fmicb.2025.1636322">10.3389/fmicb.2025.1636322</a>.
</div>
<div id="ref-megahit" class="csl-entry" role="listitem">
Li, D., Luo, R., Liu, C.-M., Leung, C.-M., Ting, H.-F., Sadakane, K., Yamashita, H. and Lam, T.-W., 2016. <span>MEGAHIT</span> v1.0: A fast and scalable metagenome assembler driven by advanced methodologies and community practices. <em>Methods</em>, 102, pp.3â€“11. https://doi.org/<a href="https://doi.org/10.1016/j.ymeth.2016.02.020">10.1016/j.ymeth.2016.02.020</a>.
</div>
<div id="ref-bwa" class="csl-entry" role="listitem">
Li, H. and Durbin, R., 2009. Fast and accurate short read alignment with burrows-wheeler transform. <em>Bioinformatics</em>, 25(14), pp.1754â€“1760. https://doi.org/<a href="https://doi.org/10.1093/bioinformatics/btp324">10.1093/bioinformatics/btp324</a>.
</div>
<div id="ref-seqkit" class="csl-entry" role="listitem">
Shen, W., Sipos, B. and Zhao, L., 2024. SeqKit2: A swiss army knife for sequence and alignment processing. <em>iMeta</em>, [online] 3(3), p.e191. https://doi.org/<a href="https://doi.org/10.1002/imt2.191">10.1002/imt2.191</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>